src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets

```diff
@@ -72,6 +72,7 @@ Copyright (c) .NET Foundation. All rights reserved.
                  '$(RuntimeIdentifier)' == '' and
                  '$(_IsExecutable)' == 'true' and '$(IsTestProject)' != 'true' and
                  '$(IsRidAgnostic)' != 'true' and
+                 '$(_IsPublishing)' == 'true' and
                   (
                     '$(SelfContained)' == 'true' or
```

Comment:
If `SelfContained` is set to true, we should automatically set the RuntimeIdentifier even if not publishing.  If there's a `PublishSelfContained` property (maybe it's in 8.0 or just planned), then in that case the RuntimeIdentifier should only be set when publishing.

Reply:
Gotcha, implemented that behavior. `PublishSelfContained` has been around for a bit now in 7.0.200, you reviewed it! 😉